app/src/main/java/com/patrickgatewood/weather/ui/WeatherPresenter.java

```diff
@@ -0,0 +1,13 @@
+package com.patrickgatewood.weather.ui;
+
+
+import android.content.Context;
+
+public interface WeatherPresenter {
+
+    void onFetchButtonClick();
+
+    void attachView(Context weatherContext);
```

Comment:
@nishtahir My intent was to lessen the restriction I'd put on attachView when I required a WeatherActivity instance. I've now realized that is not the way to accomplish what I'd wanted.